src/shared/capture/capture_basler.cpp

```diff
@@ -253,13 +250,15 @@ RawImage CaptureBasler::getFrame() {
 
 		if(grab_result->GetPayloadType() == Pylon::PayloadType_ChunkData &&
 		GenApi::IsReadable(grab_result->ChunkTimestamp)){
-		    const double period = 1.0 / (double) camera_frequency;// freq should always be 125 MHz for Basler-ace-1300-75gc
-		    long unsigned int timeStamp = grab_result->ChunkTimestamp.GetValue();
-		    double baslerTime = period* (double) timeStamp;
-		    addOffset(computerTime-baslerTime);
-		    img.setTime(getAverageOffset()+baslerTime);
-		}else{
-		    img.setTime(computerTime);
+                    uint64_t  image_timestamp = grab_result->ChunkTimestamp.GetValue();
+                    timeSync.update(image_timestamp);
+                    double time = timeSync.sync(image_timestamp) / 1e9;
```

Comment:
The time from the basler camera's is not in Nanoseconds but rather in internal system ticks, which can vary for each camera or even configuration, so you have to get it at startup using camera->GetTimestampTickFrequency.GetValue() as in line 109. The most common value here is 125 MHz, (which is the default in the header). Line 109 also doesn't compile yet on my system. I suggest, adding camera_frequency to the header again.